service/http.directory/src/main/java/org/kaazing/gateway/service/http/directory/HttpDirectoryServiceHandler.java

```diff
@@ -53,7 +53,7 @@
     private boolean indexes;
 
     private List<PatternCacheControl> patterns;
-    private static Map<String, CacheControlHandler> urlCacheControlMap = new HashMap<String, CacheControlHandler>();
+    private Map<String, CacheControlHandler> urlCacheControlMap = new ConcurrentHashMap<String, CacheControlHandler>();
```

Comment:
The map is used to match the URL of a requested file with a pattern from the configuration file. That is only done once and needs to be cached, so there is no need to remove the entries. 
The max-age and any other information that may change is computed at the request time.